### PR TITLE
Add a unique_ptr wrapper for rocksdb::Iterator

### PR DESCRIPTION
**Pull request by PragmaTwice**

### CODE REVIEW

File: src/util.h [hunk 1]
```diff
@@ -72,7 +75,17 @@ uint64_t GetTimeStampUS(void);
 // refer to https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique
 template <typename T, typename... Args>
 std::unique_ptr<T> MakeUnique(Args&& ... args) {
-     return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
+  return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
 }
 
+struct UniqueIterator : std::unique_ptr<rocksdb::Iterator> {
```
Comment (caipengbo) [inline src/util.h]: Should we reveal on the name that this is the Rocksdb iterator?

**Comment by ShooterIT** [comment]
cool, but is it suitable to put this in `util`?
actually we use many rocksdb utility methods that implement kvrocks logic.
Maybe we can create a new file, we simply package some rocksdb methods in it,
we also can easily change if we don't want to use methods provided by RocksDB,
of course, can put the function of this PR into.

And WDYT @git-hulk

**Comment by git-hulk** [comment]
> cool, but is it suitable to put this in `util`? actually we use many rocksdb utility methods that implement kvrocks logic. Maybe we can create a new file, we simply package some rocksdb methods in it, we also can easily change if we don't want to use methods provided by RocksDB, of course, can put the function of this PR into.
> 
> And WDYT @git-hulk

Yes, it will be good to put the db util functions together, this also can resolve this comment: https://github.com/apache/incubator-kvrocks/pull/650#discussion_r904428217

**Comment by PragmaTwice** [comment]
> > cool, but is it suitable to put this in `util`? actually we use many rocksdb utility methods that implement kvrocks logic. Maybe we can create a new file, we simply package some rocksdb methods in it, we also can easily change if we don't want to use methods provided by RocksDB, of course, can put the function of this PR into.
> > And WDYT @git-hulk
> 
> Yes, it will be good to put the db util functions together, this also can resolve this comment: [#650 (comment)](https://github.com/apache/incubator-kvrocks/pull/650#discussion_r904428217)

May I ask which header file those db utils are in? Is in `redis_db.h`? Thanks!

**Comment by git-hulk** [comment]
> > > cool, but is it suitable to put this in `util`? actually we use many rocksdb utility methods that implement kvrocks logic. Maybe we can create a new file, we simply package some rocksdb methods in it, we also can easily change if we don't want to use methods provided by RocksDB, of course, can put the function of this PR into.
> > > And WDYT @git-hulk
> > 
> > 
> > Yes, it will be good to put the db util functions together, this also can resolve this comment: [#650 (comment)](https://github.com/apache/incubator-kvrocks/pull/650#discussion_r904428217)
> 
> May I ask which header file those db utils are in? Is in `redis_db.h`? Thanks!

Do you think it's a good idea to create another header file to do this? like db_util.h or others.

**Comment by PragmaTwice** [comment]
> > > > cool, but is it suitable to put this in `util`? actually we use many rocksdb utility methods that implement kvrocks logic. Maybe we can create a new file, we simply package some rocksdb methods in it, we also can easily change if we don't want to use methods provided by RocksDB, of course, can put the function of this PR into.
> > > > And WDYT @git-hulk
> > > 
> > > 
> > > Yes, it will be good to put the db util functions together, this also can resolve this comment: [#650 (comment)](https://github.com/apache/incubator-kvrocks/pull/650#discussion_r904428217)
> > 
> > 
> > May I ask which header file those db utils are in? Is in `redis_db.h`? Thanks!
> 
> Do you think it's a good idea to create another header file to do this? like db_util.h or others.

Cool, done.

File: src/util.h (hunk 1 above)
Comment (PragmaTwice) [reply src/util.h]: Now it is `DBUtil::Iterator`, more clear.

**Review by git-hulk — APPROVED** [review]
LGTM, cool!

**Review by ShooterIT — APPROVED** [review]

**Comment by PragmaTwice** [comment]
Thanks everyone, merging...

**Pull request merged**